version: '3.8'

services:
  db:
    image: postgres:13
    container_name: postgres_db
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
      POSTGRES_DB: mydb
    volumes: 
      - db_data:/var/lib/postgresql/data
    networks:
      - my_network
    ports: 
      - "5432:5432"
    restart: always
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U user"]
      interval: 10s
      timeout: 5s
      retries: 5

  django_app:
    build:
      context: ./annivNovembre
    container_name: django_app
    environment:
      - POSTGRES_USER=user
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=mydb
      - POSTGRES_HOST=db
      - POSTGRES_PORT=5432
      - DJANGO_SETTINGS_MODULE=annivNovembre.settings
    volumes:
      - ./annivNovembre:/app
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    networks:
      - my_network

volumes:
  db_data:  

networks:
  my_network:
    driver: bridge
